### PR TITLE
Refactor to Typescript

### PR DESCRIPTION
**Pull request by khaosdoctor**
Refactoring to Typescript

### CODE REVIEW

**Comment by khaosdoctor** [comment]
Can we merge this?

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ node_modules
 .node_repl_history
 .nyc_output
 npm-debug.log*
+dist/
```
Comment (rwillians) [inline .gitignore]: ensure feed line

**Comment by rwillians** [comment]
hey @khaosdoctor sorry, hadn't seen it 'till now

File: .gitignore (hunk 1 above)
Comment (rwillians) [inline .gitignore:8]: insure feed line

(I'll add an .editorconfig after merging this)

File: package.json [hunk 2]
```diff
@@ -25,12 +27,20 @@
...
+  "files": [
+    "dist/**/*"
+  ],
   "license": "MIT",
   "bugs": {
     "url": "https://github.com/rwillians/express-rescue/issues"
   },
   "homepage": "https://github.com/rwillians/express-rescue#readme",
   "devDependencies": {
+    "@types/express": "^4.16.1",
```
Comment (rwillians) [inline package.json:42]: add `typescript` as dev dependency

**Review by rwillians — CHANGES_REQUESTED** [review]

**Comment by rwillians** [comment]
Meh, nevermind I'll add those changes before  publishing

**Pull request merged**